### PR TITLE
Add column object_ownership_controls in table aws_s3_bucket closes #1545

### LINKED ISSUE #1545: Add api call getBucketOwnershipControls in table `aws_s3_bucket`
Reference: 
https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/s3@v1.27.1#Client.GetBucketOwnershipControls

Add api call getObjectLockConfiguration in table `aws_s3_bucket`

image

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
➜  aws-test git:(issue-1545) ✗ ./tint.js aws_s3_bucket
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_s3_bucket []

PRETEST: tests/aws_s3_bucket

TEST: tests/aws_s3_bucket
Running terraform
data.aws_region.alternate: Reading...
data.aws_region.alternate: Read complete after 0s [id=us-east-1]
data.aws_canonical_user_id.current_user: Reading...
data.aws_partition.current: Reading...
data.aws_region.primary: Reading...
data.aws_caller_identity.current: Reading...
data.aws_partition.current: Read complete after 0s [id=aws]
data.aws_region.primary: Read complete after 0s [id=us-east-2]
data.aws_caller_identity.current: Read complete after 2s [id=123424325245]
data.null_data_source.resource: Reading...
data.null_data_source.resource: Read complete after 0s [id=static]
data.aws_canonical_user_id.current_user: Read complete after 2s [id=d1a8fb36ce85e6f4a77eec13f53958565ce292fb0c7f8fce671bb0a9eb25dd40]

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_kms_key.mykey will be created
  + resource "aws_kms_key" "mykey" {
      + arn                                = (known after apply)
      + bypass_policy_lockout_safety_check = false
      + customer_master_key_spec           = "SYMMETRIC_DEFAULT"
      + deletion_window_in_days            = 10
      + description                        = "This key is used to encrypt bucket objects"
      + enable_key_rotation                = false
      + id                                 = (known after apply)
      + is_enabled                         = true
      + key_id                             = (known after apply)
      + key_usage                          = "ENCRYPT_DECRYPT"
      + multi_region                       = (known after apply)
      + policy                             = (known after apply)
      + tags_all                           = (known after apply)
    }

  # aws_s3_bucket.named_test_resource will be created
  + resource "aws_s3_bucket" "named_test_resource" {
      + acceleration_status         = (known after apply)
      + acl                         = (known after apply)
      + arn                         = (known after apply)
      + bucket                      = "turbottest37545"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + object_lock_enabled         = (known after apply)
      + policy                      = (known after apply)
      + region                      = (known after apply)
      + request_payer               = (known after apply)
      + tags                        = {
          + "name" = "turbottest37545"
        }
      + tags_all                    = {
          + "name" = "turbottest37545"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + cors_rule {
          + allowed_headers = [
              + "*",
            ]
          + allowed_methods = [
              + "PUT",
              + "POST",
            ]
          + allowed_origins = [
              + "https://s3-website-test.hashicorp.com",
            ]
          + expose_headers  = [
              + "ETag",
            ]
          + max_age_seconds = 3000
        }

      + grant {
          + id          = "d1a8fb36ce85e6f4a77eec13f53958565ce292fb0c7f8fce671bb0a9eb25dd40"
          + permissions = [
              + "FULL_CONTROL",
            ]
          + type        = "CanonicalUser"
        }

      + lifecycle_rule {
          + enabled = true
          + id      = "log"
          + prefix  = "log/"
          + tags    = {
              + "autoclean" = "true"
              + "rule"      = "log"
            }

          + expiration {
              + days = 90
            }

          + transition {
              + days          = 30
              + storage_class = "STANDARD_IA"
            }
          + transition {
              + days          = 60
              + storage_class = "GLACIER"
            }
        }
      + lifecycle_rule {
          + enabled = true
          + id      = "tmp"
          + prefix  = "tmp/"

          + expiration {
              + date = "2022-01-12"
            }
        }

      + logging {
          + target_bucket = (known after apply)
          + target_prefix = (known after apply)
        }

      + object_lock_configuration {
          + object_lock_enabled = "Enabled"
        }

      + replication_configuration {
          + role = (known after apply)

          + rules {
              + delete_marker_replication_status = (known after apply)
              + id                               = (known after apply)
              + prefix                           = (known after apply)
              + priority                         = (known after apply)
              + status                           = (known after apply)

              + destination {
                  + account_id         = (known after apply)
                  + bucket             = (known after apply)
                  + replica_kms_key_id = (known after apply)
                  + storage_class      = (known after apply)

                  + access_control_translation {
                      + owner = (known after apply)
                    }

                  + metrics {
                      + minutes = (known after apply)
                      + status  = (known after apply)
                    }

                  + replication_time {
                      + minutes = (known after apply)
                      + status  = (known after apply)
                    }
                }

              + filter {
                  + prefix = (known after apply)
                  + tags   = (known after apply)
                }

              + source_selection_criteria {
                  + sse_kms_encrypted_objects {
                      + enabled = (known after apply)
                    }
                }
            }
        }

      + server_side_encryption_configuration {
          + rule {
              + bucket_key_enabled = (known after apply)

              + apply_server_side_encryption_by_default {
                  + kms_master_key_id = (known after apply)
                  + sse_algorithm     = (known after apply)
                }
            }
        }

      + versioning {
          + enabled    = true
          + mfa_delete = false
        }

      + website {
          + error_document           = (known after apply)
          + index_document           = (known after apply)
          + redirect_all_requests_to = (known after apply)
          + routing_rules            = (known after apply)
        }
    }

  # aws_s3_bucket_policy.b will be created
  + resource "aws_s3_bucket_policy" "b" {
      + bucket = (known after apply)
      + id     = (known after apply)
      + policy = (known after apply)
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + account_id        = "123424325245"
  + aws_partition     = "aws"
  + canonical_user_id = "d1a8fb36ce85e6f4a77eec13f53958565ce292fb0c7f8fce671bb0a9eb25dd40"
  + kms_key_id        = (known after apply)
  + resource_aka      = (known after apply)
  + resource_name     = "turbottest37545"
aws_kms_key.mykey: Creating...
aws_s3_bucket.named_test_resource: Creating...
aws_kms_key.mykey: Creation complete after 3s [id=e858abe7-07d8-41b5-b69d-aa3dead0c7ae]
aws_s3_bucket.named_test_resource: Still creating... [10s elapsed]
aws_s3_bucket.named_test_resource: Creation complete after 13s [id=turbottest37545]
aws_s3_bucket_policy.b: Creating...
aws_s3_bucket_policy.b: Creation complete after 0s [id=turbottest37545]

Warning: Deprecated

  with data.null_data_source.resource,
  on variables.tf line 48, in data "null_data_source" "resource":
  48: data "null_data_source" "resource" {

The null_data_source was historically used to construct intermediate values
to re-use elsewhere in configuration, the same can now be achieved using
locals

(and one more similar warning elsewhere)

Warning: Argument is deprecated

  with aws_s3_bucket.named_test_resource,
  on variables.tf line 59, in resource "aws_s3_bucket" "named_test_resource":
  59: resource "aws_s3_bucket" "named_test_resource" {

Use the aws_s3_bucket_versioning resource instead

(and 16 more similar warnings elsewhere)

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

account_id = "123424325245"
aws_partition = "aws"
canonical_user_id = "d1a8fb36ce85e6f4a77eec13f53958565ce292fb0c7f8fce671bb0a9eb25dd40"
kms_key_id = "arn:aws:kms:us-east-2:123424325245:key/e858abe7-07d8-41b5-b69d-aa3dead0c7ae"
resource_aka = "arn:aws:s3:::turbottest37545"
resource_name = "turbottest37545"

Running SQL query: test-get-query.sql
[
  {
    "akas": [
      "arn:aws:s3:::turbottest37545"
    ]
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "acl": {
      "Grants": [
        {
          "Grantee": {
            "DisplayName": null,
            "EmailAddress": null,
            "ID": "d1a8fb36ce85e6f4a77eec13f53958565ce292fb0c7f8fce671bb0a9eb25dd40",
            "Type": "CanonicalUser",
            "URI": null
          },
          "Permission": "FULL_CONTROL"
        }
      ],
      "Owner": {
        "DisplayName": null,
        "ID": "d1a8fb36ce85e6f4a77eec13f53958565ce292fb0c7f8fce671bb0a9eb25dd40"
      }
    },
    "bucket_policy_is_public": false,
    "logging": null,
    "name": "turbottest37545",
    "object_lock_configuration": {
      "ObjectLockEnabled": "Enabled",
      "Rule": null
    },
    "replication": null,
    "versioning_enabled": true,
    "versioning_mfa_delete": false
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "akas": [
      "arn:aws:s3:::turbottest37545"
    ],
    "bucket_policy_is_public": false,
    "logging": null,
    "name": "turbottest37545",
    "partition": "aws",
    "tags": {
      "name": "turbottest37545"
    },
    "tags_src": [
      {
        "Key": "name",
        "Value": "turbottest37545"
      }
    ],
    "title": "turbottest37545",
    "versioning_enabled": true,
    "versioning_mfa_delete": false
  }
]
✔ PASSED

TEARDOWN: tests/aws_s3_bucket

SUMMARY:

1/1 passed.

```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select name, object_ownership_controls,logging, region from aws_s3_bucket where name = 'test-delete90'
+---------------+------------------------------------------------+---------+-----------+
| name          | object_ownership_controls                      | logging | region    |
+---------------+------------------------------------------------+---------+-----------+
| test-delete90 | {"Rules":[{"ObjectOwnership":"ObjectWriter"}]} | <null>  | us-east-2 |
+---------------+------------------------------------------------+---------+-----------+
```
</details>

### CODE REVIEW

**Comment by khushboo9024** [comment]
@ParthaI  Getting below error while running list call for S3 bucket table, please check,
```
> select * from aws_s3_bucket
Error: operation error S3: GetBucketOwnershipControls, https response error StatusCode: 404, RequestID: 2Z1RtgdjkkdhHJRQ0, HostID: GI0+sdhffhshYYUIvfjkfkfwpVdcy9Igc/GEZ050jzps74aA59Is+k08yYYYUII64GHI=, api error OwnershipControlsNotFoundError: The bucket ownership controls were not found (SQLSTATE HV000)
> 
```

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by misraved — COMMENTED** [review]
@ParthaI could you please update the table docs highlighting the use of this column? Thanks!!

File: aws/table_aws_s3_bucket.go [hunk 1]
```diff
@@ -182,6 +186,13 @@ func tableAwsS3Bucket(_ context.Context) *plugin.Table {
 				Type:        proto.ColumnType_JSON,
 				Hydrate:     getObjectLockConfiguration,
 			},
+			{
+				Name:        "object_ownership_controls",
+				Description: "The OwnershipControls for an Amazon S3 bucket.",
```
Comment (misraved) [inline aws/table_aws_s3_bucket.go:191]:
```suggestion
				Description: "The Ownership Controls for an Amazon S3 bucket.",
```

File: aws/table_aws_s3_bucket.go [hunk 2]
```diff
@@ -315,6 +326,46 @@ func getS3BucketEventNotificationConfigurations(ctx context.Context, d *plugin.Q
 	return nil, nil
 }
 
+func getS3BucketObjectOwnershipControl(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// Bucket location will be nil if getBucketLocation returned an error but
+	// was ignored through ignore_error_codes config arg
```
Comment (misraved) [inline aws/table_aws_s3_bucket.go:331]: Why is this comment required? and what error do we get for bucket location?

File: docs/tables/aws_s3_bucket.md [hunk 3]
```diff
@@ -180,3 +180,14 @@ from
 where
   object_lock_configuration ->> 'ObjectLockEnabled' = 'Enabled';
 ```
+
+### List buckets object ownership control rules
```
Comment (misraved) [inline docs/tables/aws_s3_bucket.md:184]:
```suggestion
### List object ownership control rules of buckets
```

**Review by misraved — COMMENTED** [review]
@ParthaI please take a look at the review comments. Thanks!!

**Review by misraved — APPROVED** [review]

**Pull request merged**